spec/controllers/expenses_controller_spec.rb

```diff
@@ -0,0 +1,132 @@
+require "rails_helper"
+
+describe ExpensesController do
+  context "GET #new" do
+    context "when invalid params" do
+      it "won't create a new expense" do
+        user = create(:user)
+        create(:budget, user: user)
+        allow(controller).to receive(:current_user).and_return(user)
+        expense_params = { name: "Gas", bill_estimate: 500, frequency: "" }
+
+        expect do
+          post :create, params: { expense: expense_params }
+        end.to change(Expense, :count).by(0)
+      end
+
+      it "renders #new" do
+        user = create(:user)
+        create(:budget, user: user)
+        allow(controller).to receive(:current_user).and_return(user)
+        expense_params = { name: "Gas", bill_estimate: 500, frequency: "" }
+
+        post :create, params: { expense: expense_params }
+
+        expect(response).to render_template(:new)
+      end
+
+      it "sets the flash[:alert]" do
+        user = create(:user)
+        create(:budget, user: user)
+        allow(controller).to receive(:current_user).and_return(user)
+        expense_params = { name: "Gas", bill_estimate: 500, frequency: "" }
+
+        post :create, params: { expense: expense_params }
+
+        expect(flash[:alert]).
+          to match I18n.
+          t("flash.actions.create.alert", resource_name: "Expense")
+      end
+    end
+
+    context "when there isn't a budget" do
+      it "won't create a new expense" do
```

Comment:
Something to consider: do we want to automagically create a budget when a user is created?
It makes sense, because this app will not be here without a budget. 
So we can create a default budget for a user, and thus don't need to worry about these specs.
Maybe for another PR?

Also, since a user has one budget now, what is the point of the Budget class? Maybe something to ponder and discuss later

Reply:
That's a lot to ponder! Creating a budget with a user seems sound and the budget class doesn't have a huge role, definitely things to consider in another PR!